File: projects/Go/examples/serialization.go
package main

import "fmt"
import "../proto/fbe"
import "../proto/proto"

func main() {
	// Create a new account with some orders
	var account = proto.NewAccountFromFieldValues(1, "Test", proto.State_good, proto.Balance{Currency: "USD", Amount: 1000.0}, &proto.Balance{Currency: "EUR", Amount: 100.0}, make([]proto.Order, 0))
	account.Orders = append(account.Orders, proto.Order{Id: 1, Symbol: "EURUSD", Side: proto.OrderSide_buy, Type: proto.OrderType_market, Price: 1.23456, Volume: 1000.0})
	account.Orders = append(account.Orders, proto.Order{Id: 2, Symbol: "EURUSD", Side: proto.OrderSide_sell, Type: proto.OrderType_limit, Price: 1.0, Volume: 100.0})
	account.Orders = append(account.Orders, proto.Order{Id: 3, Symbol: "EURUSD", Side: proto.OrderSide_buy, Type: proto.OrderType_stop, Price: 1.5, Volume: 10.0})

	// Serialize the account to the FBE stream
	writer := proto.NewAccountModel(fbe.NewEmptyBuffer())
	if _, err := writer.Serialize(account); err != nil {
		panic("serialization error")
	}
	if ok := writer.Verify(); !ok {
		panic("verify error")
	}

	// Show the serialized FBE size
	fmt.Printf("FBE size: %d\n", writer.Buffer().Size())

	// Deserialize the account from the FBE stream
	reader := proto.NewAccountModel(writer.Buffer())
	if ok := reader.Verify(); !ok {
		panic("verify error")
	}
	account, _, err := reader.Deserialize()
	if err != nil {
		panic("deserialization error")
	}

	// Show account content
	fmt.Println()
	fmt.Println(account)
}

File: projects/Kotlin/src/examples/Serialization.kt
package examples

import java.util.*

import com.chronoxor.proto.*
import com.chronoxor.proto.fbe.*

object Serialization
{
    @JvmStatic
    fun main(args: Array<String>)
    {
        // Create a new account with some orders
        val account = Account(1, "Test", State.good, Balance("USD", 1000.0), Balance("EUR", 100.0), ArrayList())
        account.orders.add(Order(1, "EURUSD", OrderSide.buy, OrderType.market, 1.23456, 1000.0))
        account.orders.add(Order(2, "EURUSD", OrderSide.sell, OrderType.limit, 1.0, 100.0))
        account.orders.add(Order(3, "EURUSD", OrderSide.buy, OrderType.stop, 1.5, 10.0))

        // Serialize the account to the FBE stream
        val writer = AccountModel()
        writer.serialize(account)
        assert(writer.verify())

        // Show the serialized FBE size
        println("FBE size: " + writer.buffer.size)

        // Deserialize the account from the FBE stream
        val reader = AccountModel()
        reader.attach(writer.buffer)
        assert(reader.verify())
        reader.deserialize(account)

        // Show account content
        println()
        println(account)
    }
}
